### PR TITLE
SNAPSHOTS+TESTS: Correctly Wait for Clean State

### LINKED ISSUE #36779: [CI] SnapshotDisruptionIT. testDisruptionOnSnapshotInitialization fails unreproducibly
See https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/837/console

Ran this with hundred iterations on Linux and once on osx, does not reproduce. Reproduction line is

```
 ./gradlew :server:integTest \
  -Dtests.seed=642A19B3B0D37154 \
  -Dtests.class=org.elasticsearch.discovery.SnapshotDisruptionIT \
  -Dtests.method="testDisruptionOnSnapshotInitialization" \
  -Dtests.security.manager=true \
  -Dtests.locale=hr-HR \
  -Dtests.timezone=Europe/Tirane \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

I saw another test failure in the last days. Log excerpt

```
  1> [2018-12-18T14:58:52,986][INFO ][o.e.d.SnapshotDisruptionIT] [testDisruptionOnSnapshotInitialization] after test
FAILURE 4.73s J1 | SnapshotDisruptionIT.testDisruptionOnSnapshotInitialization <<< FAILURES!
   > Throwable #1: java.lang.AssertionError:
   > Expected: an instance of org.elasticsearch.discovery.MasterNotDiscoveredException
   >      but: <RemoteTransportException[[node_tm1][127.0.0.1:41899][cluster:admin/snapshot/create]]; nested: ConcurrentSnaps
hotExecutionException[[test-repo:test-snap-2] cannot snapshot while a snapshot deletion is in-progress];> is a org.elasticsea
rch.transport.RemoteTransportException
   >    at __randomizedtesting.SeedInfo.seed([642A19B3B0D37154:4EF53F6396CB828B]:0)
   >    at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   >    at org.elasticsearch.discovery.SnapshotDisruptionIT.testDisruptionOnSnapshotInitialization(SnapshotDisruptionIT.java:
164)
   >    at java.lang.Thread.run(Thread.java:748)

```

Direct log https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/837/consoleText

### PR DESCRIPTION
**Pull request by original-brownbear**
* Test must wait for no in progress deletion as well here since the master failover leads to a snapshot deletion
* Closes #36779

------------------

This is pretty easy to reproduce by putting a small sleep (5s or so) in e.g. the task spun off by `org.elasticsearch.snapshots.SnapshotsService#deleteSnapshotFromRepository` as well.

### CODE REVIEW

**Comment by original-brownbear** [comment]
Jenkins test this

File: server/src/test/java/org/elasticsearch/discovery/SnapshotDisruptionIT.java [hunk 1]
```diff
@@ -135,9 +135,14 @@ public void clusterChanged(ClusterChangedEvent event) {
         assertBusy(() -> {
             SnapshotsInProgress snapshots = dataNodeClient().admin().cluster().prepareState().setLocal(false).get().getState()
                 .custom(SnapshotsInProgress.TYPE);
+            SnapshotDeletionsInProgress snapshotDeletionsInProgress =
+                dataNodeClient().admin().cluster().prepareState().setLocal(false).get().getState().custom(SnapshotDeletionsInProgress.TYPE);
```
Comment (ywelsch) [inline server/src/test/java/org/elasticsearch/discovery/SnapshotDisruptionIT.java]: `local = false` is the default. No need to specify this here (also remove from one line above). Perhaps only fetch the cluster state once

**Review by ywelsch — APPROVED** [review]
LGTM

**Review by tlrx — APPROVED** [review]

**Comment by original-brownbear** [comment]
Jenkins run default-distro tests

**Pull request merged**